From 6b2526cc8ea5b03b4b3c13dcc35544e92b88b4ae Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 11 Apr 2022 05:44:12 +0300
Subject: [PATCH] [SW-66675] rom: update unaligned api implementation

Implementation of "get_unaligned_be32()" api in rom is aligned
with other repos (preboot, uboot) .

Change-Id: I8ee8eb760bcc5731b56664a6b57d2cada53bd4cb
---

diff --git a/common/rsa/hl_rsa_arc.h b/common/rsa/hl_rsa_arc.h
index 9b68a2d..1c71ede 100644
--- a/common/rsa/hl_rsa_arc.h
+++ b/common/rsa/hl_rsa_arc.h
@@ -9,7 +9,7 @@
 
 #define put_unaligned_be32(a, b) (*(uint32_t *)(b) = __builtin_bswap32(a))
 
-static inline uint32_t get_unaligned_be32(const uint32_t *ptr)
+static inline uint32_t get_unaligned_be32(const void *ptr)
 {
 	const uint8_t *p = (const uint8_t *)ptr;
 	return p[0] << 24 | p[1] << 16 | p[2] << 8 | p[3];
